### ISSUE TITLE
broken snapshot value when className === tag name

### ISSUE THREAD

**Issue by cameronmcefee** [styled-components/jest-styled-components#48]
Hi there! I think I may have found a bug. Given the following component:

```
import styled from 'styled-components';

const TextArea = styled.textarea.attrs({
  className: 'textarea',
  spellCheck: 'false'
})``;

export default TextArea;
```
And the following test:

```
import React from 'react';
import { shallow } from 'enzyme';
import TextInput from './TextInput';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';

describe('TextInput', function() {
  test('renders', () => {
    const wrapper = shallow(<TextInput />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
```

the following output will be generated:

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders 1`] = `


<  className="bDWFJH"
  spellCheck="false"
/>
`;
```
Note the missing element name in the snapshot. This appears to happen when the `className` matches the tag type. I tested with text inputs (classname "input") and textareas.

To confirm it was a conflict of class/tag, I tried changing the className to 'foo', which caused the tag name to appear correctly in the output.

To try to rule out styled-components or react itself (in case I was doing some sort of worst practice by doing this), I removed jest-styled-components from my tests. Without it, changing the `className` value had no noticeable impact on the output, so my best guess is that the issue is here.

**Comment by MicheleBertoli** [comment]
Thank you very much @cameronmcefee.
Good catch, and perfect information + repro steps in the issue.

This is fixed in 4.0.1:
```js
exports[`TextArea renders 1`] = `
<textarea
  className=""
  spellCheck="false"
/>
`;
```

**Comment by MicheleBertoli** [comment]
In v4.1.0, the custom class name is preserved as well:

```js
exports[`TextArea renders 1`] = `
<textarea
  className="textarea "
  spellCheck="false"
/>
`;
```
